### ISSUE TITLE
angular + user-event@14: Chip no longer added to mat-chip-list using userEvent.type(input, 'Something{Enter}')

### ISSUE THREAD

**Issue by enolive** [testing-library/user-event#1065]
### Reproduction example

https://codesandbox.io/p/github/enolive/mat-chips-testing/draft/angry-ully

### Prerequisites

1. Setup a new Angular project with angular/material, jest, testing-library
2. Copy the Chips Input Example from [Angular Material](https://material.angular.io/components/chips/overview#chip-input) into the main component
3. Write a test that tries to enter a new chip

```ts
it('supports adding new chips', async () => {
    await renderComponent()
    const chipsInput = screen.getByLabelText('Favorite Fruits')

    await userEvent.type(chipsInput, 'Banana{Enter}')

    const fruits = screen.getAllByRole('option')
    expect(fruits).toHaveLength(4)
  })
```

4. Execute the test

### Expected behavior

As a new chip was entered, the list should now contain a new element making the total length 4.

### Actual behavior

The list has still only the default 3 elements. 

I can also prove by adding some log that the event `matChipInputTokenEnd` for adding a new chip is no longer fired. The keyup event still gets the Enter key pressed.

### User-event version

14.4.3

### Environment

Testing Library framework:  `@testing-library/angular@12.1.2` 

JS framework:  `angular@14.2.0`

Test environment: `jest@28.1.3` 

DOM implementation:  `jsdom@19.0.0`

### Additional context

When I downgrade to user-event@13 (and remove the await from the userEvent.type call since it is a new feature in 14), everything works

**Comment by ph-fritsche** [comment]
Could you break down which exact behavior is reported as a bug here? Please try to create a minimal example showcasing the problem without third-party code.
Is this a duplicate of #982  ?

**Comment by enolive** [comment]
For some reason, my Codesandbox is no longer available so I created a new:

https://codesandbox.io/p/github/enolive/mat-chips-testing/draft/angry-ully

> Could you break down which exact behavior is reported as a bug here? Please try to create a minimal example showcasing the problem without third-party code. Is this a duplicate of #982 ?

The shown component should add newly typed tags to a list on enter:

![Before Adding a Banana](https://user-images.githubusercontent.com/20992515/202257783-40982b7e-e0bc-4184-b4c9-2a63ba52761f.png)

![After Adding a Banana](https://user-images.githubusercontent.com/20992515/202254128-400a9c8e-cfb9-41d5-bfd7-c90ae384a369.png)

Unfortunately, the function that is triggered on enter (add) is not triggered at all when running the tests

[add()](https://codesandbox.io/p/github/enolive/mat-chips-testing/draft/angry-ully?file=%2Fsrc%2Fapp%2Fapp.component.ts&selection=%5B%7B%22endColumn%22%3A4%2C%22endLineNumber%22%3A19%2C%22startColumn%22%3A4%2C%22startLineNumber%22%3A19%7D%5D)

The result when running `npm test` is

```log
 FAIL  src/app/app.component.spec.ts (7.525 s)
  AppComponent
    ✓ renders heading (379 ms)
    ✓ displays the favorite fruits (150 ms)
    ✕ supports adding new chips (399 ms)
    ✓ has access to DOM (7 ms)

  ● AppComponent › supports adding new chips

    expect(received).toHaveLength(expected)

    Expected length: 4
    Received length: 3
```

This means that only the 3 default fruits are in the list, but not my newly added fruit "Banana"

The codesandbox is the smallest example I can come up with. 

I have no clue why the error occurs, the same test runs with userevent@13 and no 3rd-party-lib changes without any problems, so it is hard for me to decide if this is the same problem as in the bug you mentioned

**Comment by enolive** [comment]
I also noticed that using {Tab} instead of {Enter} actually works. 

But I would like to stay as close to the real user interaction in my tests as possible.

**Comment by enolive** [comment]
after some debugging (including angular material). I can confirm that this is indeed a duplicate of #982. 

this is the line in angular material that is responsible for the problem:

https://github.com/angular/components/blob/14.2.x/src/material/chips/chip-input.ts#L239

as you can see, it is indeed the usage of keyCode, which was deprecated in user-event@13

following workaround (until a fix in material) works:

```ts
it('supports adding new chips', async () => {
    await renderComponent()
    const chipsInput = screen.getByLabelText('Favorite Fruits')

    await userEvent.type(chipsInput, 'Banana')
    fireEvent.keyDown(chipsInput, {keyCode: 13})

    const fruits = screen.getAllByRole('option')
    expect(fruits).toHaveLength(4)
  })
```

**Comment by dzhavat** [comment]
@enolive Oh, man! Thanks for your workaround. I used a whole day yesterday in debugging failing tests until I figured that it's the `keyCode`. I tried making the patch function a bit more generic, so a list of keyCodes shouldn't be necessary. But the `keyCode` is all over the Material, so I don't see this patch function becoming obsolete any time soon 😬

```js
function patchKeyEvent(e: KeyboardEvent) {
  Object.defineProperty(e, 'keyCode', {
    get: () => e.key.charCodeAt(0),
  });
}

document.addEventListener('keydown', patchKeyEvent, { capture: true });
```
